File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        char[] chars = br.readLine().toCharArray();
        Stack<Character> stack = new Stack<>();
        int sum = 0;

        for (int i = 0; i < chars.length; i++) {
            if (chars[i] == '(') stack.push('(');
            else {
                stack.pop();
                if (chars[i - 1] == '(') sum += stack.size();
                else sum += 1;
            }
        }
        System.out.println(sum);
    }
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.kt
import java.io.*
import java.util.Stack

fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    val line = readLine().split("")
    val stack = Stack<Char>()
    var sum = 0

    for (i in line.indices) {
        if (line[i] == "(") stack.push('(')
        else if (line[i] == ")"){
            stack.pop()
            sum += if (line[i-1] == "(") {
                stack.size
            } else {
                1
            }
        }
    }
    println(sum)
}